sys/auto_init/screen/auto_init_lvgl.c

```diff
@@ -48,8 +48,10 @@ void auto_init_lvgl(void)
     LOG_DEBUG("[auto_init_screen] initializing lvgl\n");
 
     /* Only a single screen is supported by lvgl */
+#if !IS_USED(MODULE_LV_DRIVERS_DISPLAY_MONITOR)
```

Comment:
I stared at this for a while thinking that parts of the Initialisation of `DISPLAY_MONITOR` should go here or everything should move to `lvgl_init()`.

same goes for that `s_screen` it is just used in this one function and than i is moved to `contrib/lvgl.c` but just as a pointer but there it is used in mutiple functions. 
Seems to me as if `s_screen` should be on the stack and copied ( it is also small).
But at the same time `screen_dev.h` is marked experimental maybe there is other future use i cant see

Reply:
> same goes for that `s_screen` it is just used in this one function and than i is moved to `contrib/lvgl.c` but just as a pointer but there it is used in mutiple functions.
> Seems to me as if `s_screen` should be on the stack and copied ( it is also small).
> But at the same time `screen_dev.h` is marked experimental maybe there is other future use i cant see

Maybe @aabadie has more insights.

Reply:
I think it's important to keep the screen variable instance out of lvgl since a board setup could provide mutilple screens. Having screen as a parameter of the `lvgl_init` function allows the user to choose what screen will be used by lvgl. Another screen could be used for something else.
It's common with auto_init code, to have static instances of entity defines in the auto_init code, even if they are only used once.